distribution/engine/THIRD-PARTY/NOTICE

```diff
@@ -356,6 +356,126 @@ The license file can be found at `licenses/BSD-3-Clause`.
 Copyright notices related to this dependency can be found in the directory `org.jline.jline-3.21.0`.
 
 
+'org-netbeans-api-annotations-common', licensed under the The Apache Software License, Version 2.0, is distributed with the engine.
```

Comment:
Wow, that is a lot of dependencies. Some of them seem pretty obsolete and should be reduced on the NetBeans side. Anyway for our purposes, we want to use just the `sampler.jar` without any dependencies. We should basically:

```scala
libraryDependencies ++= Seq(
  "org.netbeans.api" % "org-netbeans-modules-sampler" % "RELEASE130",
  exclude("org.netbeans.api", "org-openide-loaders") exclude("org.netbeans.api", "....")
)
```

Reply:
How do you know what are the required dependencies for the `org-netbeans-modules-sampler` and what can be excluded?

Reply:
> How do you know what are the required dependencies ... and what can be excluded?

If it continues to run, it can be excluded ;-) On a serious note: it is tricky. On the other hand, I helped to architect the module, and I know we wanted to have it as independent on NetBeans as possible.

I've just went through the code in https://github.com/apache/netbeans/pull/4047 and cleaned up as many references as possible while keeping the NetBeans IDE functionality untouched. Version `NETBEANS140` shall have significantly less dependencies.

For a standalone usage we can remove even more: when I remove the remaining ones:

![obrazek](https://user-images.githubusercontent.com/26887752/165768615-31a8f852-0acc-47ff-a83f-a1d53f119996.png)

The compilation errors appear only in `InternalSampler` and `SelfSampleVFS` - we are not using these two classes - e.g. I believe Enso code should run fine without the dependencies. Then there is _"Common Annotations"_ but that is only used for `@NonNull` & co. annotations which do not affect runtime either.

Anyway, having a test in CI to verify the sampling works in Enso is good idea in any case.

Reply:
Improved in [3fd211c](https://github.com/enso-org/enso/pull/3389/commits/3fd211cf318448a0bb03a67e97c4219b78e4595e) - I'll make sure it gets even better with NetBeans 14.0